### PR TITLE
transport: add UDP support

### PR DESCRIPTION
**Pull request by GlassyYang**
add UDP face so python-ndn can connect NDN forward plane by UDP.
UDP face is best fit for small, unreliable and low lantecy data transport.

### CODE REVIEW

File: src/ndn/transport/stream_socket.py
Comment (yoursunny) [inline src/ndn/transport/stream_socket.py:106]: UDP is not a *stream* socket, so that it does not belong in `stream_socket.py`.

* Move the `Face` type to a new `face.py` file.
* Create a separate file for `UdpFace` type.

File: src/ndn/transport/stream_socket.py
Comment (yoursunny) [inline src/ndn/transport/stream_socket.py:151]: Format your code with `autopep8`.

File: src/ndn/client_conf.py
Comment (yoursunny) [inline src/ndn/client_conf.py:111]: This will not work correctly with literal IPv6 address.
Use `urllib.parse`, https://stackoverflow.com/a/53172593

**Review by yoursunny — CHANGES_REQUESTED** [review]

File: src/ndn/transport/face.py
Comment (yoursunny) [inline src/ndn/transport/face.py:151]: Format your code according to PEP8.
You can use a tool like `autopep8`.

File: src/ndn/transport/face.py
Comment (yoursunny) [inline src/ndn/transport/face.py:104]: Keep StreamFace and its subclasses in `stream_face.py` for backwards compatibility.
Put UdpFace in a new file `udp_face.py`.
The Face type should be in `face.py`, but re-exported through `stream_face.py` for backwards compatibility.

**Review by yoursunny — CHANGES_REQUESTED** [review]

**Comment by GlassyYang** [comment]
thank you for your suggestions.
I have finished them all.
@yoursunny

File: src/ndn/transport/face.py
Comment (yoursunny) [inline src/ndn/transport/face.py:110]: Declare the defaults 127.0.0.1 and 6363 on this line.
Get rid of the `if` statements.
The member declarations in above three lines are also unnecessary.

File: src/ndn/transport/face.py
Comment (yoursunny) [inline src/ndn/transport/face.py:21]: Sort imports with `isort`.
Moreover, you should add autopep8 and isort into CI scripts to prevent future mistakes.

File: src/ndn/transport/udp_face.py
Comment (yoursunny) [inline src/ndn/transport/udp_face.py:28]: Declare the defaults 127.0.0.1 and 6363 on this line.
Get rid of the `if` statements.
The member declarations in above three lines are also unnecessary.

File: tests/face_test.py
Comment (yoursunny) [inline tests/face_test.py:3]: Sort imports with `isort`.
Moreover, you should add autopep8 and isort into CI scripts to prevent future mistakes.

**Review by yoursunny — CHANGES_REQUESTED** [review]

**Review by yoursunny — APPROVED** [review]

**Comment by yoursunny** [comment]
There's a build error detected by GitHub Actions.
@GlassyYang  please fix it with a follow-up Pull Request.
@zjkmxy  you should have approved GitHub Actions run (approval needed for first time contributor) before merging, so that such problem is detected before merging.

**Pull request merged**
